core/src/main/scala/com/lightbend/paradox/ParadoxProcessor.scala

```diff
@@ -78,25 +81,122 @@ class ParadoxProcessor(reader: Reader = new Reader, writer: Writer = new Writer)
+    ignorePaths:      List[Regex],
+    validateAbsolute: Boolean,
+    logger:           ParadoxLogger): Int = {
+
+    val errorCollector = new ErrorCollector
+
+    val roots = parsePages(mappings, identity, properties, errorCollector)
+    val pages = Page.allPages(roots)
+    val globalPageMappings = rootPageMappings(roots)
+    val fullSite = allSiteFiles.map(_.swap).toMap
+
+    val linkCapturer = new LinkCapturer
+
+    @tailrec
+    def validate(location: Option[Location[Page]]): Unit = location match {
+      case Some(loc) =>
+        val page = loc.tree.label
+        val pageProperties = properties ++ page.properties.get
+        val currentMapping = Path.generateTargetFile(Path.relativeLocalPath(page.rootSrcPage, page.file.getPath), globalPageMappings)
+        val writerContext = Writer.Context(loc, pages, reader, writer, new PagedErrorContext(errorCollector, page),
+          logger, currentMapping, "", "", groups, pageProperties)
+        val serializer = linkCapturer.serializer(writerContext)
+        page.markdown.accept(serializer)
+        validate(loc.next)
+      case None => ()
+    }
+
+    roots.foreach { root =>
+      validate(Some(root.location))
+    }
+
+    def ignore(path: String) = ignorePaths.exists(_.pattern.matcher(path).matches())
+
+    linkCapturer.allLinks.foreach {
+      case CapturedLinkWithSources(CapturedAbsoluteLink(uri), sources) if validateAbsolute && !ignore(uri.toString) =>
+        validateExternalLink(uri, sources, errorCollector, logger)
+      case CapturedLinkWithSources(CapturedRelativeLink(path, fragment), sources) if !ignore(path) =>
+        fullSite.get(path) match {
+          case Some(file) =>
+            fragment.foreach { f =>
+              validateFragment(path, Source.fromFile(file, "UTF-8").mkString, f, sources, errorCollector)
+            }
+          case None =>
+            sources.foreach {
+              case (file, node) => errorCollector(s"Could not find path [$path] in site", file, node)
+            }
+
+        }
+      case _ =>
+      // Ignore
+    }
+
+    errorCollector.logErrors(logger)
+    errorCollector.errorCount
+  }
+
+  private def validateExternalLink(uri: URI, sources: List[(File, Node)], errorContext: ErrorContext, logger: ParadoxLogger) = {
+    logger.info(s"Validating external link: $uri")
+    val conn = uri.toURL.openConnection().asInstanceOf[HttpURLConnection]
+    conn.addRequestProperty("User-Agent", "Paradox Link Validator <https://github.com/lightbend/paradox>")
+    try {
+      if (conn.getResponseCode != 200) {
+        sources.foreach {
+          case (file, node) => errorContext(s"Error validating external link, status code was ${conn.getResponseCode}", file, node)
+        }
+      } else {
+        if (uri.getFragment != null) {
+          val content = Source.fromInputStream(conn.getInputStream).mkString
+          validateFragment(uri.toString, content, uri.getFragment, sources, errorContext)
+        }
+      }
+    } finally {
```

Comment:
Trying out the external link validation on the Cinnamon docs, got a failure (actual exceptions rather than any error reporting). Could be useful to catch exceptions here and collect as errors?

(It was an SSLHandshakeException "unable to find valid certification path" on one of the https links)

Reply:
Updated to catch non fatal errors and report.